engine/src/scheme.rs

```diff
@@ -183,6 +183,37 @@ impl<'i> Display for ParseError<'i> {
     }
 }
 
+macro_rules! declare_config {
+    ($($(#[$attr:meta])* let $name:ident : $type:ty = $value:expr),*) => {
+        /// Repository of all configuration options.
+        #[derive(Deserialize)]
+        pub struct SchemeConfig {
+            $($(#[$attr])* pub $name: $type),*
+        }
+
+        impl SchemeConfig {
+            /// Creates a new scheme configuration repository
+            /// with default values.
+            pub fn new() -> Self {
+                Self {
+                    $($name: $value),*
+                }
+            }
+        }
+
+        impl Default for SchemeConfig {
+            fn default() -> Self {
+                Self::new()
+            }
+        }
+    };
+}
+
+declare_config!(
+    #[doc="Call stack depth limit"]
+    let call_stack_depth_limit: usize = std::usize::MAX
```

Comment:
In this particular case usage of macro seems to be an overkill: everything can be easily and clearly expressed using standard language syntax. Declare fields in structure, set default values in `Default` trait implementation.